arch/xtensa/src/esp32/esp32_efuse_lowerhalf.c

```diff
@@ -0,0 +1,189 @@
+
+struct esp32_efuse_lowerhalf_s
+{
+  FAR const struct efuse_ops_s *ops; /* Lower half operations */
+  void *upper;                       /* Pointer to efuse_upperhalf_s */
+};
+
+/****************************************************************************
+ * Private Functions Prototypes
+ ****************************************************************************/
+
+/* "Lower half" driver methods **********************************************/
+
+static int      esp32_efuse_read_field(FAR struct efuse_lowerhalf_s *lower,
+                                       const efuse_desc_t *field[],
+                                       FAR uint8_t *data, size_t size);
+static int      esp32_efuse_write_field(FAR struct efuse_lowerhalf_s *lower,
+                                        const efuse_desc_t *field[],
+                                        FAR const uint8_t *data,
+                                        size_t size);
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/* "Lower half" driver methods */
+
+static const struct efuse_ops_s g_esp32_efuse_ops =
+{
+  .read_field   = esp32_efuse_read_field,
+  .write_field  = esp32_efuse_write_field,
+  .ioctl      = NULL,
+};
+
+/* EFUSE lower-half */
+
+static struct esp32_efuse_lowerhalf_s g_esp32_efuse_lowerhalf =
+{
+  .ops = &g_esp32_efuse_ops,
+  .upper = NULL,
+};
+
+/****************************************************************************
+ * Private functions
+ ****************************************************************************/
+
+static int esp32_efuse_read_field(FAR struct efuse_lowerhalf_s *lower,
+                                  const efuse_desc_t *field[],
+                                  uint8_t *data, size_t bits_len)
+{
+  int ret = OK;
+
+  /* Read the requested field */
+
+  ret = esp_efuse_read_field(field, data, bits_len);
+
+  return ret;
+}
+
+static int esp32_efuse_write_field(FAR struct efuse_lowerhalf_s *lower,
+                                   const efuse_desc_t *field[],
+                                   const uint8_t *data, size_t bits_len)
+{
+  irqstate_t flags;
+  int ret = OK;
+
+  flags = enter_critical_section();
+
+  /* Write the blob data to the field */
+
+  ret = esp_efuse_write_field(field, data, bits_len);
+
+  /* Burn the EFUSEs */
+
+  esp_efuse_burn_efuses();
+
+  leave_critical_section(flags);
+
+  return ret;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: efuse_ioctl
+ ****************************************************************************/
+
+static int efuse_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
```

Comment:
this is a private function inside the Public functions section, move to the previous seciton